server/events/project_command_output_wrapper.go

```diff
@@ -29,22 +29,24 @@ func (p *ProjectOutputWrapper) updateProjectPRStatus(commandName command.Name, c
 	// Create a PR status to track project's plan status. The status will
 	// include a link to view the progress of atlantis plan command in real
 	// time
-	if err := p.JobURLSetter.SetJobURLWithStatus(ctx, commandName, models.PendingCommitStatus); err != nil {
+	statusId, err := p.JobURLSetter.SetJobURLWithStatus(ctx, commandName, models.PendingCommitStatus)
+	if err != nil {
 		ctx.Log.Errorf("updating project PR status", err)
 	}
+	ctx.StatusId = statusId
```

Comment:
mutating objects that have already been constructed is a slippery slope, is this necessary? Can we avoid doing this in anyway? Even something like:

```go
type SomeNewObject struct {
  ProjectCommandContext
  StatusId string
}

```

is better.  Mutating objects makes it hard for new developers to track where a field is set and how it's used and can lead to outages from wrong assumptions.